src/main/java/net/dv8tion/jda/api/interactions/modals/ModalMapping.java

```diff
@@ -69,32 +72,48 @@ public Component.Type getType()
      *
      * <p>For TextInputs, this returns what the User typed in it.
      *
+     * @throws IllegalStateException
+     *         If the value of this {@link ModalMapping} cannot be represented as a String.
+     *
      * @return The String representation of this component.
      */
     @Nonnull
     public String getAsString()
     {
-        return value;
+        if (object.isNull("value") || !object.isType("value", DataType.STRING))
+            typeError("String");
+        return object.getString("value");
     }
 
-    @Override
-    public String toString()
+    /**
+     * The {@link List} representation of this component.
+     *
+     * <p>For SelectMenus, this returns the values of all the options the user selected.
+     *
+     * @throws IllegalStateException
+     *         If the value of this {@link ModalMapping} cannot be represented as a List of Strings.
+     *
+     * @return The {@link List} representation of this component.
+     */
+    @Nonnull
+    public List<String> getAsStringList()
     {
-        return "ModalMapping[" + getType() + "](" + getId() + "=" + getAsString() + ")";
+        if (object.isNull("values") || !object.getArray("values").isType(0, DataType.STRING))
+            typeError("List<String>");
+
+        return object.getArray("values")
+                .stream(DataArray::getString)
+                .collect(Collectors.toList());
     }
 
     @Override
-    public boolean equals(Object o)
```

Comment:
Is there a reason why equals was removed?

Reply:
`DataObject` doesn't override `equals()` and this might get messy otherwise. (Especially in the future if Discord adds more types). But i can just add it back and compare the types manually